CRM/Contribute/Tokens.php

```diff
@@ -55,21 +56,33 @@ public function getCurrencyFieldName() {
    */
   protected function getRelatedTokens(): array {
     $tokens = [];
-    if (!in_array('ContributionRecur', array_keys(\Civi::service('action_object_provider')->getEntities()))) {
-      return $tokens;
-    }
-    $hiddenTokens = ['modified_date', 'create_date', 'trxn_id', 'invoice_id', 'is_test', 'payment_token_id', 'payment_processor_id', 'payment_instrument_id', 'cycle_day', 'installments', 'processor_id', 'next_sched_contribution_date', 'failure_count', 'failure_retry_date', 'auto_renew', 'is_email_receipt', 'contribution_status_id'];
-    $contributionRecurFields = ContributionRecur::getFields(FALSE)->setLoadOptions(TRUE)->execute();
-    foreach ($contributionRecurFields as $contributionRecurField) {
-      $tokens['contribution_recur_id.' . $contributionRecurField['name']] = [
-        'title' => $contributionRecurField['title'],
-        'name' => 'contribution_recur_id.' . $contributionRecurField['name'],
+
+    if (in_array('ContributionPage', array_keys(\Civi::service('action_object_provider')->getEntities()))) {
+      $contributionPageReceiptText = ContributionPage::getFields(FALSE)->setLoadOptions(TRUE)->addWhere('name', '=', 'receipt_text')->execute()->first();
+      $tokens['contribution_page_id.' . $contributionPageReceiptText['name']] = [
+        'title' => $contributionPageReceiptText['title'],
+        'name' => 'contribution_page_id.' . $contributionPageReceiptText['name'],
         'type' => 'mapped',
-        'options' => $contributionRecurField['options'] ?? NULL,
-        'data_type' => $contributionRecurField['data_type'],
-        'audience' => in_array($contributionRecurField['name'], $hiddenTokens) ? 'hidden' : 'user',
+        'options' => $contributionPageReceiptText['options'] ?? NULL,
```

Comment:
I can't think why we would need to define options? (I see it is defined on the previously existing one - perhaps it is needed for some fields?)

Reply:
Removed